compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1041,6 +1048,57 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
         checker
     }
 
+    pub(super) fn register_predefined_opaques_in_new_solver(&mut self) {
```

Comment:
how is that fine? :thinking: aren't the preregistered opaque types in the wrong environment now, they have the generic params of the opaque while in the context of the defining scope?

Reply:
as a thought which also deals with https://github.com/rust-lang/trait-system-refactor-initiative/issues/17

long term we could maybe end up with the following to replace `DefiningAnchor`:
```rust
enum DefineOpaques {
    // the current `DefiningAnchor::Bind`
    Yes(&'tcx List<DefId>),
    // We do not allow any new definitions for our opaques,
    // but when using an opaque with `DefId`, we instantiate the
    // inferred hidden type with these substs.
    //
    // This doesn't quite work because of lifetimes, but we can
    // figure something out there :grin:
    No(SomeMapCollection<DefId, EarlyBinder<Ty<'tcx>>),
}
```
anyways, your PR works for RPIT so I wouldn't mind us landing this as long as we put a fixme here.

Reply:
You're absolutely right lol. We shouldn't map RPITITs back to their declaration params during writeback if we're in the new trait solver. 🤦 

I'll add a fixme anyways.